lib/AST/ASTPrinter.cpp

```diff
@@ -2805,10 +2824,14 @@ void PrintAST::visitFuncDecl(FuncDecl *decl) {
 
       Printer.printDeclResultTypePre(decl, ResultTyLoc);
       Printer.callPrintStructurePre(PrintStructureKind::FunctionReturnType);
-      if (decl->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>())
+      if (decl->getAttrs().hasAttribute<ImplicitlyUnwrappedOptionalAttr>()) {
         printTypeLocForImplicitlyUnwrappedOptional(ResultTyLoc);
-      else
-        printTypeLoc(ResultTyLoc);
+      } else {
+        PrintOptions subOptions = Options;
+        subOptions.OpaqueReturnTypePrinting =
+            PrintOptions::OpaqueReturnTypePrintingMode::WithOpaqueKeyword;
+        printTypeLocWithOptions(ResultTyLoc, subOptions);
```

Comment:
Missing the "HACK" comment here?

Also, what happens when we support optional-wrapped opaque types? Can we handle both cases?

Reply:
I suppose we could just SaveAndRestore this setting for both implicitly unwrapped optional opaque result types and regular types. Wouldn't hurt, and would Just Work™ when we get around to it